plugins/tests/greatexpectations/test_schema.py

```diff
@@ -0,0 +1,176 @@
+import os
+import sqlite3
+import tempfile
+
+import pandas as pd
+import pytest
+from flytekitplugins.greatexpectations import BatchRequestConfig, GEConfig, GEType
+from great_expectations.exceptions import ValidationError
+
+from flytekit import task, workflow
+
+
+def test_ge_schema_task():
+    ge_config = GEConfig(
+        data_source="data",
+        expectation_suite="test.demo",
+        data_connector="data_example_data_connector",
+    )
+
+    @task
+    def get_length(dataframe: pd.DataFrame) -> int:
+        return dataframe.shape[0]
+
+    @task
+    def my_task(csv_file: GEType[ge_config]) -> pd.DataFrame:
```

Comment:
is it that great expectations is limited to csv files or other types of files only?
Can we also use primitive types?

Maybe this is a question for Abe.

Reply:
I wouldn't say it's restricted to files. For suppose, take MySQL. Here, GE expects a table name. dataset pertains to `data_asset_name` in GE which is always a string. 

Check this out: https://docs.greatexpectations.io/en/latest/autoapi/great_expectations/core/batch/index.html?highlight=data_asset_name#great_expectations.core.batch.BatchRequest